### PR TITLE
DOC: EX01 (part of ExtensionArray)

### PR DESCRIPTION
**Pull request by Charlie-XIAO**
- [x] Towards #37875

`ExtensionArray` is kind of tricky, I don't know what kind of example should we put here. Currently I'm providing some (not directly runnable) example implementation for each method. Alternatively we can provide a fully implemented example in `ExtensionArray`, and link to that example in each method. Please let me know if I should make any modification.

### CODE REVIEW

**Comment by Charlie-XIAO** [comment]
Sure, I'll re-do this PR.

File: pandas/core/arrays/base.py [hunk 1]
```diff
@@ -504,14 +504,22 @@ def to_numpy(
...
         """
         raise AbstractMethodError(self)
 
     @property
     def shape(self) -> Shape:
         """
         Return a tuple of the array dimensions.
+
+        Since Extension Arrays are only allowed to be 1-dimensional,
+        the shape would be ``(len(self),)``.
```
Comment (MarcoGorelli) [inline pandas/core/arrays/base.py:522]: this isn't always true, can you remove it please?

Comment (Charlie-XIAO) [reply pandas/core/arrays/base.py:522]: Oh thanks for pointing out. I forgot to remove this when cleaning up.

**Review by MarcoGorelli — APPROVED** [review]
looks good to me, thanks @Charlie-XIAO

**Pull request merged**